icenet.c

```diff
@@ -397,6 +467,15 @@ static int icenet_start_xmit(struct sk_buff *skb, struct net_device *ndev)
 	ndev->stats.tx_packets++;
 	ndev->stats.tx_bytes += skb->len;
 
+	if (send_comp_avail(nic) > 32) {
```

Comment:
The most recent revision of the internal NAPI driver was tuned differently.  Perhaps this partially accounts for the performance discrepancy.

```suggestion
	if (send_comp_avail(nic) > 8) {
```

Reply:
Didn't even notice this one. It should really be a configuration option.

Reply:
Ideally, it should be a function of the queue depth and some platform characteristic(s) dicovered at runtime (e.g., core count).  It's too low-level to be a user-facing parameter, and the current threshold is probably over-optimized for iperf3 bandwidth.

While it is a blunt way to avoid Tx interrupts entirely, coupling the Tx completion handling with the NAPI poll could be risky.  Interrupt mitigation would work better as a hardware feature with programmable IRQ watermarks.